Mathport/Syntax/Transform/Tactic.lean

```diff
@@ -25,3 +33,11 @@ mathport_rules
         let tacs' := tacs[0:i-1] ++ #[tacs[i].setArg 0 tac'] ++ tacs[i+1:tacs.size]
         return Syntax.node info ``Parser.Tactic.tacticSeq1Indented #[Syntax.node info2 `null tacs']
     throwUnsupported
+
+-- common obsolete patterns from haveI
+mathport_rules
+  | `(by have $hd:haveDecl; exact $t) => `(have $hd:haveDecl; $t)
+  | `(by have $hd:haveDecl <;> exact $t) => `(have $hd:haveDecl; $t)
```

Comment:
I wonder whether we should have a general rule saying `by $tac1 <;> $tac2` => `by $tac1; $tac2`. Mathlib is ambiguous about which one is used, but I suspect it is more often the latter than the former.

Reply:
This is not correct in general, e.g. when we have `by cases h <;> simp`.  For `have` it's clear that we get exactly one goal.  This line also goes one step further and converts the have to term-mode syntax.